test/unit/Ramm/swap.js

```diff
@@ -1,87 +1,146 @@
 
-    const ethOut = liquidity.sub(liquidity.mul(newNXM).div(newNXM.add(amount)));
+    const nxmIn = parseEther('1');
+    const state = await getReserves(fixture.state, pool, tokenController, nextBlockTimestamp);
+    const currentLiquidity = state.eth;
 
-    const timestamp = await ramm.lastSwapTimestamp();
-    await nxm.connect(member).approve(tokenController.address, amount);
+    await nxm.connect(member).approve(tokenController.address, nxmIn);
 
+    // before state
+    const totalSupplyBefore = await tokenController.totalSupply();
     const nxmBalanceBefore = await nxm.balanceOf(member.address);
     const ethBalanceBefore = await ethers.provider.getBalance(member.address);
 
-    await setNextBlockTime(timestamp.add(6 * 60 * 60).toNumber());
-    const tx = await ramm.connect(member).swap(amount);
-    const { effectiveGasPrice, cumulativeGasUsed } = await tx.wait();
+    const tx = await ramm.connect(member).swap(nxmIn, parseEther('0.015')); // initial sportPriceB 0.0152
+    const { gasUsed, effectiveGasPrice } = await tx.wait();
 
+    // after state
+    const totalSupplyAfter = await tokenController.totalSupply();
     const nxmBalanceAfter = await nxm.balanceOf(member.address);
     const ethBalanceAfter = await ethers.provider.getBalance(member.address);
-
-    expect(nxmBalanceBefore).to.be.equal(nxmBalanceAfter.add(amount));
-    expect(ethBalanceBefore).to.be.equal(ethBalanceAfter.sub(ethOut).add(effectiveGasPrice.mul(cumulativeGasUsed)));
+    const stateAfter = await getState(ramm);
+
+    // expected state
+    const newNxmB = state.nxmB.add(nxmIn);
+    const newLiquidity = currentLiquidity.mul(state.nxmB).div(newNxmB);
+    const newNxmA = state.nxmA.mul(newLiquidity).div(currentLiquidity);
+    const ethOut = currentLiquidity.sub(newLiquidity);
+
+    expect(totalSupplyAfter).to.be.equal(totalSupplyBefore.sub(nxmIn));
+    expect(nxmBalanceAfter).to.be.equal(nxmBalanceBefore.sub(nxmIn));
+    expect(ethBalanceAfter).to.be.equal(ethBalanceBefore.add(ethOut).sub(gasUsed.mul(effectiveGasPrice)));
+
+    expect(stateAfter.nxmA).to.be.equal(newNxmA);
+    expect(stateAfter.nxmB).to.be.equal(newNxmB);
+    expect(stateAfter.eth).to.be.equal(newLiquidity);
+    expect(stateAfter.timestamp).to.be.equal(nextBlockTimestamp);
   });
 
-  it('should swap ETH for NXM', async function () {
+  it.only('should swap ETH for NXM', async function () {
     const fixture = await loadFixture(setup);
-    const { ramm, nxm, tokenController } = fixture.contracts;
+    const { ramm, nxm, pool, tokenController } = fixture.contracts;
     const [member] = fixture.accounts.members;
 
-    const amount = parseEther('1');
-    const liquidity = BigNumber.from('2050000000000000000000');
-    const newNXM = BigNumber.from('68826162646107933349888');
+    const { timestamp } = await ethers.provider.getBlock('latest');
+    const nextBlockTimestamp = timestamp + 6 * 60 * 60;
+    await setNextBlockTime(nextBlockTimestamp - 1);
```

Comment:
block.timestamp on swap is always 1ms ahead of what I set in setNextBlockTime.

not really sure why or how to solve this